.ado/pipelines/templates/steps-frontdoor-traffic-switch.yaml

```
@@ -158,26 +176,76 @@ steps:
+      $privateLinkInfraDeployOutput = Get-ChildItem $(Pipeline.Workspace)/terraformOutputPrivateLinkInfra/*.json | Get-Content | ConvertFrom-JSON
 
-      if($LastExitCode -ne 0)
+      # loop through stamps from pipeline artifact json
+      foreach($stamp in $privateLinkInfraDeployOutput.stamp_properties.value)
       {
-          throw "*** Error on purging cache in Front Door $(frontdoor_name)"
-      }
\ No newline at end of file
+        $plServiceId = $stamp.private_link_service_resource_id
+        echo "*** Looking up Private Endpoint connections on Private Link $plServiceId"
+
+        $peConnections = az network private-endpoint-connection list --id $plServiceId | ConvertFrom-Json
+
+        # Get pending connections which were made by our Front Door
+        $afdConnections = $peConnections | Where-Object { $_.properties.privateLinkServiceConnectionState.description -eq "Request access for CDN Frontdoor Private Link Origin for prefix $(prefix)$(globalResourceSuffix)" }
+        if(-not $afdConnections)
+        {
+          # If there are no connections from Front Door, that indicates a problem as we would expect at least a pending connection at this point
+          throw "*** No Private Endpoint connections from Front Door on Private Link service $plServiceId"
+        }
+
+        # Get connections which are pending approval
+        $pendingConnections = $afdConnections | Where-Object { $_.properties.privateLinkServiceConnectionState.status -eq "Pending" }
+
+        if(-not $pendingConnections)
+        {
+          echo "*** Found $($afdConnections.Count) Private Endpoint connections from Front Door, all of which are already approved"
+        }
+
+        foreach($pendingPe in $pendingConnections)
+        {
+          echo "*** Found pending Private Endpoint Connection from Front Door: $($pendingPe.name). Approving..."
+          az network private-endpoint-connection approve --id $pendingPe.id `
+                                                         --description "$($pendingPe.properties.privateLinkServiceConnectionState.description)" # keep original description
+
+          if($LastExitCode -ne 0)
+          {
+              throw "*** Error on approving private endpoint connection $($pendingPe.name)"
+          }
+        }
+      }
+
+- ${{ if eq(parameters.purgeAfdCache, 'true') }}:
+  # After new backends have been added to Front Door, we purge the cache so that content will be retrieved from the new backends as well
+  - task: AzureCLI@2
+    displayName: 'Purge Front Door cache'
+    retryCountOnTaskFailure: 1
+    inputs:
+      azureSubscription: $(azureServiceConnection)
+      scriptType: pscore
+      scriptLocation: inlineScript
+      inlineScript: |
+
+        echo "*** Purging Front Door cache with path /*"
+        az afd endpoint purge -g $(global_resource_group_name) --profile-name $(frontdoor_name) --content-paths "/*" --endpoint-name "$(prefix)$(globalResourceSuffix)-primaryendpoint"
```

```suggestion
        az afd endpoint purge `
        --resource-group $(global_resource_group_name) `
        --profile-name $(frontdoor_name) `
        --content-paths "/*" `
        --endpoint-name "$(prefix)$(globalResourceSuffix)-primaryendpoint"
```

for easier readability